scripts/generate_test_data.py

```diff
@@ -0,0 +1,35 @@
+import os
+import yaml
+import argparse as ap
+from csv_data_service import csv_data_service
+from concurrent.futures import ProcessPoolExecutor
+
+
+def main():
+    parser = ap.ArgumentParser()
+    parser.add_argument("-c", "--config")
+    args = parser.parse_args()
+
+    with open(os.path.expanduser(args.config), "r") as fo:
+        config = yaml.load(fo, Loader=yaml.SafeLoader)
+
+    dir_path = config["csv_directory"]
+    data_columns = config["columns"]
+    range_of_days = config["max_days"]
+    num_of_participants = config["participant_count"]
+
+    if not os.path.isdir(dir_path):
+        os.mkdir(dir_path)
+
+    csv_service = csv_data_service.CSVDataService(dir_path, data_columns, range_of_days)
+    csv_service.prepare_csv_test_data()
+
+    with ProcessPoolExecutor(4) as exe:
+        _ = [
+            exe.submit(csv_service.write_test_csv_files(i), i)
+            for i in range(num_of_participants)
+        ]
```

Comment:
I've since updated it. [d97e827](https://github.com/AMP-SCZ/dpimport/pull/23/commits/d97e827d189ddb13ab68452e3835da395657c592)
but what is happening is, I'm setting the number of cores to use, most machines have 4. 
The map functions first argument is the task we want to run, then the second argument are the items to run the task with. Then behind the scenes this is managing the threads to be used and will pick up the next task in parallel as they become finish/become available.